src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs

```diff
@@ -230,6 +234,19 @@ public partial class UpdateDurableFunction
                     // Cleanup the new version when the update activity fails
                     await TryCleanupActivityV2(context, instanceWatermarks, input.Partition);
                 }
+
+                if (!isFailedToUpdateStudy)
+                {
+                    await context.CallActivityWithRetryAsync(
+                     nameof(DeleteOldVersionBlobV2Async),
+                     _options.RetryOptions,
+                     new CleanupBlobArguments(instanceWatermarks, input.Partition));
+
+                    await context.CallActivityWithRetryAsync(
```

Comment:
This is probably not actually a breaking change as the activities are at the end... I think lol. Although is `DeleteOldVersionBlobV2Async` executing now in a different circumstance?

Reply:
Yeah, I found a bug. Not sure how it missed. DeleteOldVersionBlobV2Async was only called for last studyInstanceUId. But it should be called for every commit.

Reply:
Also we don't have delete the old version blobs if the current activity run returns an error.